pkg/rebalancer/storage/controller/service/rebalancer.go

```diff
@@ -353,13 +404,25 @@ func (r *rebalancer) getPodByAgentName(agentName string) (*pod.Pod, error) {
 }
 
 func (r *rebalancer) createJob(ctx context.Context, jobTpl job.Job, reason config.RebalanceReason, agentName, agentNs string, rate float64) error {
-	// check indexing or not
 	if reason == config.DEVIATION {
 		p, err := r.getPodByAgentName(agentName)
 		if err != nil {
 			return err
 		}
-		c, err := agent.New(agent.WithAddrs(p.IP))
+
+		ctx, cancel := context.WithCancel(ctx)
+		defer cancel()
+
+		agentAddr := net.JoinHostPort(p.IP, uint16(r.agentPort))
+
+		gc := grpc.New(grpc.WithAddrs(agentAddr), grpc.WithConnectionPoolSize(1), grpc.WithInsecure(true), grpc.WithResolveDNS(false))
```

Comment:
I think `grpc.WithConnectionPoolSize(1)` and ` grpc.WithInsecure(true)` and `grpc.WithResolveDNS(false)` are used in all grpc calls.
so Should we pre-generate options from these values ​​in the constructor?